src/Database/Element/Structure.php

```diff
@@ -2,16 +2,125 @@
 
 namespace Phoenix\Database\Element;
 
+use Phoenix\Exception\StructureException;
 use Phoenix\Database\Element\MigrationTable;
 
 class Structure
 {
     /** @var Table[] */
     private $tables = [];
 
+    /**
+     * @param MigrationTable $migrationTable
+     * @return MigrationTable
+     * @throws StructureException - if some error occured - e.g. table which should be created already exists
+     */
+    public function prepare(MigrationTable $migrationTable)
```

Comment:
change name to validate, prepare can be next method when we will cange MigrationTable if needed